src/main/java/org/prebid/server/bidder/yandex/YandexBidder.java

```diff
@@ -0,0 +1,238 @@
+        }
+    }
+
+    private HttpRequest<BidRequest> buildHttpRequest(BidRequest outgoingRequest, String url) {
+        return HttpRequest.<BidRequest>builder()
+                .method(HttpMethod.POST)
+                .uri(url)
+                .headers(headers(outgoingRequest))
+                .body(mapper.encodeToBytes(outgoingRequest))
+                .payload(outgoingRequest)
+                .build();
+    }
+
+    private static MultiMap headers(BidRequest bidRequest) {
+        final MultiMap headers = HttpUtil.headers();
+        final Device device = bidRequest.getDevice();
+        if (device != null) {
+            HttpUtil.addHeaderIfValueIsNotEmpty(headers, HttpUtil.ACCEPT_LANGUAGE_HEADER, device.getLanguage());
+            HttpUtil.addHeaderIfValueIsNotEmpty(headers, HttpUtil.USER_AGENT_HEADER, device.getUa());
+            HttpUtil.addHeaderIfValueIsNotEmpty(headers, HttpUtil.X_FORWARDED_FOR_HEADER, device.getIp());
+            HttpUtil.addHeaderIfValueIsNotEmpty(headers, "X-Real-Ip", device.getIp());
+        }
+
+        return headers;
+    }
+
+    @Override
+    public Result<List<BidderBid>> makeBids(BidderCall<BidRequest> httpCall, BidRequest bidRequest) {
+        try {
+            final BidResponse bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+            return Result.of(extractBids(bidResponse, httpCall.getRequest().getPayload()), Collections.emptyList());
+        } catch (DecodeException | PreBidException e) {
+            return Result.withError(BidderError.badServerResponse(e.getMessage()));
+        }
+    }
+
+    private static List<BidderBid> extractBids(BidResponse bidResponse, BidRequest bidRequest) {
+        final List<SeatBid> seatBids = bidResponse != null ? bidResponse.getSeatbid() : null;
+        if (seatBids == null) {
+            return Collections.emptyList();
+        }
+
+        if (seatBids.isEmpty()) {
+            throw new PreBidException("SeatBids is empty");
+        }
+        return bidsFromResponse(bidResponse, bidRequest.getImp());
+    }
+
+    private static List<BidderBid> bidsFromResponse(BidResponse bidResponse, List<Imp> imps) {
+        return bidResponse.getSeatbid().stream()
+                .filter(Objects::nonNull)
+                .map(SeatBid::getBid)
+                .filter(Objects::nonNull)
+                .flatMap(Collection::stream)
+                .map(bid -> BidderBid.of(bid, getBidType(bid.getImpid(), imps), bidResponse.getCur()))
+                .collect(Collectors.toList());
+    }
+
+    private static BidType getBidType(String bidImpId, List<Imp> imps) {
+        for (Imp imp : imps) {
+            if (bidImpId.equals(imp.getId())) {
+                return resolveImpType(imp);
+            }
+        }
+        throw new PreBidException(("Invalid bid imp ID #%s does not match any imp IDs from the original "
+                + "bid request").formatted(bidImpId));
```

Comment:
Nitpick. No need of having two strings
this will be ok
```java
throw new PreBidException(("Invalid bid imp ID #%s does not match any imp IDs from the original bid request")
                .formatted(bidImpId));
```